CotEditor/Sources/Document.swift

```diff
@@ -337,6 +337,8 @@ final class Document: NSDocument, AdditionalDocumentPreparing, EncodingHolder {
             ?? Defaults[.syntaxStyle]
         
         self.setSyntaxStyle(name: styleName)
+        
+        ScriptManager.shared.dispatchEvent(documentOpened: self)
```

Comment:
Nice, i did not sure about that.
I also suggest `document resumed` event but not in this PR 😄